### PR TITLE
0.2.1 rabbitmq

### PR DESCRIPTION
**Pull request by joshuacox**
Added 0.2.1 to accomodate upstream changes https://github.com/rancher/catalog-dockerfiles/pull/103 that PR is probably obsolete with this

### CODE REVIEW

**Comment by joshuacox** [comment]
One question I have is why was the rabbitmq directory deleted from [here](https://github.com/rancher/catalog-dockerfiles)?  Did I just go through some work that was already done?

**Comment by joshuacox** [comment]
I tried to test this with the Examples folder I just added, but on my setup at least the externally linked rabbitMQ fails with errors like this:

```
7/28/2017 8:29:10 AMTypeError: Cannot read property 'createChannel' of undefined
7/28/2017 8:29:10 AM    at /home/rabbit/src/receive.js:6:7
7/28/2017 8:29:10 AM    at /home/rabbit/node_modules/amqplib/callback_api.js:16:10
7/28/2017 8:29:10 AM    at Socket.<anonymous> (/home/rabbit/node_modules/amqplib/lib/connect.js:152:18)
7/28/2017 8:29:10 AM    at Object.onceWrapper (events.js:316:30)
7/28/2017 8:29:10 AM    at emitOne (events.js:115:13)
7/28/2017 8:29:10 AM    at Socket.emit (events.js:210:7)
7/28/2017 8:29:10 AM    at emitErrorNT (internal/streams/destroy.js:64:8)
7/28/2017 8:29:10 AM    at _combinedTickCallback (internal/process/next_tick.js:138:11)
7/28/2017 8:29:10 AM    at process._tickCallback (internal/process/next_tick.js:180:9)
7/28/2017 8:29:24 AM/home/rabbit/src/receive.js:6
7/28/2017 8:29:24 AM  conn.createChannel(function(err, ch) {
7/28/2017 8:29:24 AM      ^
```

but if you do not specify the external link it uses go templating like you did with the reactioncommerce to change the docker-compose.yml

File: templates/rabbitmq-3/2/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+RabbitMQ 3.6 with easy Rancher clustering
+===
+Provides RabbitMQ image that can scale to a cluster.
+
+The following environment variables are passed to confd in order to set up RabbitMQ's configuration file:
+
+* Partition handling: RabbitMQ's cluster handling setting: default set to autoheal
+* Erlang cookie: cookie to allow nodes communication: default set to defaultcookiepleasechange
```
Comment (rawmind0) [inline templates/rabbitmq-3/2/README.md]: Syntax error...
```
-* Erlang cookie: cookie to allow nodes communication: default set to defaultcookiepleasechange
+* Erlang cookie: cookie to allow nodes communication: default set to defaultcookie please change.
```

File: templates/rabbitmq-3/2/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+catalog:
+  name: "Rabbit MQ"
+  version: "0.2.1"
+  decription: A reliable, distributed messaging broker
+  questions:
+    - variable: "erlang_cookie"
+      description: "Erlang cookie required for auth"
+      label: "Erlang cookie:"
+      required: true
+      default:
```
Comment (rawmind0) [inline templates/rabbitmq-3/2/rancher-compose.yml]: Set default erlang_cookie value??
```
-      default:
+      default: "defaultcookie" 
```

**Comment by rawmind0** [comment]
Hi @joshuacox ....Thanks for the commits....

Please, review the simple change request with syntax error.

Otherwise, rabbitmq-examples doesn't have docker-compose file....Could you please, use this PR just for rabbitmq package and a new one just for rabbitmq-examples??

**Comment by rawmind0** [comment]
I forgot.... Make sense to make the scale configurable???

File: templates/rabbitmq-3/2/README.md (hunk 1 above)
Comment (joshuacox) [inline templates/rabbitmq-3/2/README.md]: maybe this should even be a 'password' type so people can hit the generate button, and make required=true and default=nil to force people to give it a value that has no default

**Comment by rawmind0** [comment]
Thanks @joshuacox ....

LGTM...

**Comment by tomasvinduska** [comment]
Hello dude,nice try,but its not backward compatible you miss line without it its not working like previose version (sadly)

containers getting connection refused, balancer for managment returning 503

`rabbitmq-base:
  net: "container:rabbitmq"`

**Comment by rawmind0** [comment]
Hi @tomasvinduska ...Thanks so much for the advise...

Please try with new merged version.

**Comment by joshuacox** [comment]
Let me know if there are still issues I'll submit another PR and fix it ASAP

**Pull request merged**